orchestrator/relayer/src/fee_manager.rs

```diff
@@ -0,0 +1,196 @@
+                    return true
+                }
+                let token_price = match self.get_token_price(&batch_fee.token_contract_address).await {
+                    Ok(token_price) => token_price,
+                    Err(_) => return false,
+                };
+
+                let estimated_fee = estimated_cost.get_total();
+                let batch_value = batch_fee.amount * token_price;
+
+                info!("estimate cost is {}, batch value is {}", estimated_fee, batch_value);
+                return batch_value >= estimated_fee
+            }
+            RelayerMode::Api => {
+                let body = json!({
+                    "batchFee": {
+                        "amount": batch_fee.amount.to_string(),
+                        "tokenContractAddress": batch_fee.token_contract_address
+                    },
+                    "estimatedCost": {
+                        "gas": estimated_cost.gas.to_string(),
+                        "gasPrice": estimated_cost.gas_price.to_string()
+                    }}
+                );
+
+                return match Client::new()
+                    .post(self.relayer_api_url.as_str())
+                    .json(&body)
+                    .send().await {
+                    Ok(resp) =>
+                        match resp.json().await {
+                        Ok(json) => {
+                            let api_response: ApiResponse = json;
+                            return if api_response.can_send {
+                                true
+                            } else {
+                                // code 5 means that it is not profitable but limit has not been
+                                // exceeded or no addresses are blacklisted
+                                // in that case we check if we should send at non profitable cost
+                                api_response.reason_type == 5 &&
+                                    self.should_send_at_non_profitable_cost(contract_address)
+                            }
+                        }
+                        Err(err) => {
+                            error!("error on relayer api: {}", err);
+                            false
+                        }
+                    }
+                    Err(err) => {
+                        error!("error on relayer api: {}", err);
+                        false
+                    }
+                }
+            }
+        }
+    }
+
+    fn should_send_at_non_profitable_cost(
+        &mut self,
+        contract_address: &EthAddress,
+    ) -> bool {
+        match self.next_batch_send_time.get(contract_address) {
+            Some(time) => {
+                return *time < Instant::now()
+            }
+            None => self.update_next_batch_send_time(*contract_address),
+        }
+        return true
+    }
+
+    pub fn update_next_batch_send_time(
+        &mut self,
+        contract_address: EthAddress,
```

Comment:
does it need to be `pub fn` or `pub (crate) fn` or `fn` is enough?

Reply:
changing to `fn` causes an error because it is uses in `batch_relaying`,  `pub (crate) fn` looks ok